src/database/type-transforms.tsx

```diff
@@ -57,14 +59,37 @@ export const buildFacility = (
 ): Facility => {
   const documentData = document.data();
 
-  let facility: Facility = documentData;
-  facility.id = document.id;
-  facility.scenarioId = scenarioId;
-  facility.createdAt = timestampToDate(documentData.createdAt);
-  facility.updatedAt = timestampToDate(documentData.updatedAt);
-  facility.modelInputs = buildModelInputs(documentData.modelInputs);
+  let {
+    createdAt,
+    updatedAt,
+    modelInputs,
+    name,
+    description,
+    systemType,
+  } = documentData;
+  const id = document.id;
+  createdAt = timestampToDate(documentData.createdAt);
```

Comment:
I might not be following, but why are we extracting `createdAt` and `updatedAt` and `modelInputs` out of `documentData` only to then both overwrite them and to reference them through `documentData` in these calls? Why couldn't we, for example, do this:
```
  let {
    name,
    description,
    systemType,
  } = documentData;
  const id = document.id;
  const createdAt = timestampToDate(documentData.createdAt);
  const updatedAt = timestampToDate(documentData.updatedAt);
  const modelInputs = buildModelInputs(documentData.modelInputs);
  const modelVersions: ModelInputs[] = [];

  return {
      createdAt,
      description,
      id,
      modelInputs,
      name,
      scenarioId,
      systemType,
      updatedAt,
      modelVersions,
    };
```

Reply:
yeah this was just a sloppy rewrite, my intention was to remove the extra references to `documentData` but apparently I only got halfway there. i'll fix this up!